src/stream-tcp-reassemble.c

```diff
@@ -1081,12 +1088,15 @@ int StreamTcpReassembleAppLayer (ThreadVars *tv, TcpReassemblyThreadCtx *ra_ctx,
                         "(base %u, isn %u, last_ack %u => diff %u) p %"PRIu64,
                         stream->base_seq, stream->isn, stream->last_ack,
                         stream->last_ack - (stream->isn + ackadd), p->pcap_cnt);
+                gap = stream->last_ack - (stream->isn + ackadd);
```

Comment:
There is an issue when we hit this case. The seg_list continues to be NULL, so its never passed to the app-layer. Not sure yet where I should be looking here.